workspaces/mcp-integrations/plugins/backstage-mcp-tools-backend/src/plugin.ts

```diff
@@ -156,42 +92,88 @@ Example invocations and the output from those invocations:
-                          'The kind/type of the Backstage entity (e.g., Component, API, System)',
-                        ),
-                      tags: z
-                        .array(z.string())
-                        .describe(
-                          'The tags associated with the Backstage entity',
-                        ),
-                      description: z
-                        .string()
-                        .optional()
-                        .describe('The description of the Backstage entity'),
-                      type: z
-                        .string()
-                        .optional()
-                        .describe(
-                          'The type of the Backstage entity (e.g., service, library, website)',
-                        ),
-                    }),
+                    z.union([
+                      z.object({
+                        name: z
+                          .string()
+                          .describe(
+                            'The name field for each Backstage entity in the catalog',
+                          ),
+                        kind: z
+                          .string()
+                          .describe(
+                            'The kind/type of the Backstage entity (e.g., Component, API, System)',
+                          ),
+                        tags: z
+                          .array(z.string())
+                          .describe(
+                            'The tags associated with the Backstage entity',
+                          ),
+                        description: z
+                          .string()
+                          .optional()
+                          .describe('The description of the Backstage entity'),
+                        type: z
+                          .string()
+                          .optional()
+                          .describe(
+                            'The type of the Backstage entity (e.g., service, library, website)',
+                          ),
+                        owner: z
+                          .string()
+                          .optional()
+                          .describe(
+                            'The owner of the Backstage entity (e.g., team-platform, user:john.doe)',
+                          ),
+                        lifecycle: z
+                          .string()
+                          .optional()
+                          .describe(
+                            'The lifecycle of the Backstage entity (e.g., production, staging, development)',
+                          ),
+                        dependsOn: z
+                          .array(z.string())
+                          .optional()
+                          .describe(
+                            'List of entities this entity depends on (e.g., component:default/database)',
+                          ),
+                      }),
+                      z.custom<Entity>(),
```

Comment:
We lose the nice described schema with the verbose output (and the context usage is _considerable_), but it's advantageous for scenarios where it doesn't make sense for us to include it in the reduced output (e.g. fields specific to only one entity kind)

Reply:
ACK yeah let's wait for the explicit req from a more specialized client